client-internal/src/main/java/org/terracotta/angela/client/support/junit/AngelaRule.java

```diff
@@ -84,7 +84,8 @@ protected void before(Description description) throws Throwable {
       }
     }
 
-    this.clusterFactory = new ClusterFactory(description.getTestClass().getSimpleName(), configuration);
+    int hash = description.getMethodName() == null ? 0 : description.getMethodName().hashCode();
+    this.clusterFactory = new ClusterFactory(description.getTestClass().getSimpleName() + "-" + hash, configuration);
```

Comment:
@mathieucarbou instead of using the hashCode of the method, we could add the method name itself, if it's available. This would help us in identifying work dir per test, which we can't do currently.

Reply:
I did this change. Also, rebased the PR - if it looks good to you, you can merge.

Reply:
Adding the method name can break tests on windows because of long paths....
The hashcode was really enough because Angela is already appending a time. So this is just to avoid collision during the same minute.
Can you revert it back ?

Reply:
@saurabhagas : I have pushed back my initial commit and rebased.